luaui/barwidgets.lua

```diff
@@ -494,33 +505,21 @@ function widgetHandler:LoadWidget(filename, fromZip, enableLocalsAccess)
 		return nil
 	end
 
-	local knownInfo = self.knownWidgets[name]
+	local knownInfo = self.knownWidgetInfos[name]
 	if knownInfo then
 		if knownInfo.active then
 			Spring.Echo('Failed to load: ' .. basename .. '  (duplicate name)')
 			return nil
 		end
+		widgetInfo = knownInfo
 	else
-		-- create a knownInfo table
-		knownInfo = {}
-		knownInfo.desc = widget.whInfo.desc
-		knownInfo.author = widget.whInfo.author
-		knownInfo.basename = widget.whInfo.basename
-		knownInfo.filename = widget.whInfo.filename
-		knownInfo.fromZip = fromZip
-		self.knownWidgets[name] = knownInfo
+		widgetInfo.fromZip = fromZip
+		self.knownWidgetInfos[name] = widgetInfo
 		self.knownCount = self.knownCount + 1
 		self.knownChanged = true
 	end
-	knownInfo.active = true
-
-	if widget.GetInfo == nil then
-		Spring.Echo('Failed to load: ' .. basename .. '  (no GetInfo() call)')
-		return nil
-	end
-
-	-- Get widget information
-	local info = widget:GetInfo()
+	widgetInfo.active = true
```

Comment:
Is there a case where it's not already active?

Reply:
this is one of the many leftovers from the previous code
the active field on the widgetInfo is not set until that line

Reply:
Oh, oops, I actually meant to leave this comment on another line. Added a comment there.